File: 2025_05/leetcode2025-05-11.py
from typing import List

class Solution:
    def threeConsecutiveOdds(self, arr: List[int]) -> bool:
        arr = [i%2 for i in arr]
        n = len(arr)
        for i in range(1,n-1):
            if arr[i-1]+arr[i]+arr[i+1]==3:
                return True
        return False

if __name__ == "__main__":
    s = Solution()
    arr = [1,2,34,3,4,5,7,23,12]
    print(s.threeConsecutiveOdds(arr))

File: 2025_05/leetcode2025-05-11.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    bool threeConsecutiveOdds(vector<int>& arr) {
        int n = arr.size();
        for (int i=1; i<n-1; i++){
            if (arr[i-1]%2+arr[i]%2+arr[i+1]%2 == 3){
                return true;
            }
        }
        return false;
    }
};

int main(){
    Solution s;
    vector<int> arr = {1,2,34,3,4,5,7,23,12};
    cout << s.threeConsecutiveOdds(arr) << endl;
}
